### PR TITLE
feat: implement smart controls GraphQL mutations

### PR DESCRIPTION
**Pull request by Veldkornet**
# Summary
This pull request implements the required GraphQL mutations for the Frank Energie API library to support reading and writing smart control states (smart charging, smart trading, smart feed-in, smart HVAC, and EV deadline settings).

# Why this is needed
To allow the Home Assistant integration (and other programmatic consumers of the library) to securely interact with and control these smart features directly rather than forcing users to use the mobile application. This establishes the necessary API primitives for bidirectional smart home control.

# Key Changes
- Added GraphQL mutation definitions for:
  - `DisableSmartTrading`
  - `SmartFeedInDisable`
  - `SmartHvacDisable`
  - `SmartHvacUpdateSettings`
  - `EnodeUpdateVehicleChargeSettings`
  - `EnodeUpdateChargerChargeSettings`
  - `EnodeEnableSmartCharging`
  - `EnodeDisableSmartCharging`
- Added library API stubs for future/extended feature support:
  - `SmartBatteryUpdateSettings`
  - `EnodeUpdateAllVehicleChargeSettings`
  - `Logout`
  - `UpdateUserSettings`
- Added comprehensive unit tests in `tests/test_smart_controls_mutations.py` to ensure all query payloads are correctly generated and mock responses are parsed accurately.

## Summary by Sourcery

Voeg GraphQL-mutatiet ondersteuning toe voor smart controls en gerelateerde accountbewerkingen, inclusief bijbehorende high-level API-methoden en tests.

Nieuwe functionaliteit:
- Stel async API-methoden beschikbaar om Enode smart charging in en uit te schakelen en om laadinstellingen van voertuigen en laders bij te werken.
- Stel async API-methoden beschikbaar om smart trading, smart feed-in en smart HVAC uit te schakelen en om smart HVAC-instellingen bij te werken.
- Introduceer async API-stubs voor smart battery-instellingen, bulkupdates van voertuiglaadinstellingen, uitloggen en updates van gebruikersinstellingen.

Tests:
- Voeg uitgebreide unit tests toe die succes-, fout-, authenticatie- en payload-vormscenario’s dekken voor alle smart control-mutatiemethoden en -stubs.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add GraphQL mutation support for smart controls and related account operations, including corresponding high-level API methods and tests.

New Features:
- Expose async API methods to enable and disable Enode smart charging and update vehicle and charger charge settings.
- Expose async API methods to disable smart trading, smart feed-in, and smart HVAC, and to update smart HVAC settings.
- Introduce async API stubs for smart battery settings, bulk vehicle charge setting updates, logout, and user settings updates.

Tests:
- Add comprehensive unit tests covering success, error, authentication, and payload-shape scenarios for all smart control mutation methods and stubs.

</details>

## Summary by CodeRabbit

* **New Features**
  * New smart controls: disable smart trading/feed‑in, EV smart charging (enable/disable), smart HVAC and battery settings, vehicle charge profile updates (including bulk), logout and user settings management.

* **Bug Fixes / Improvements**
  * Stronger authentication gating and improved GraphQL error/response handling; more robust smart battery parsing and clearer validation error messages.

* **Tests**
  * Added comprehensive async test suite covering smart controls, auth gating, mutations and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/HiDiHo01/python-frank-energie/pull/80?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: python_frank_energie/authentication.py [hunk 1]
```diff
@@ -74,11 +71,7 @@ def auth_token_valid(self, tz: timezone = UTC) -> bool:
             True if the token has not expired; otherwise, False.
         """
         try:
-            decoded = jwt.decode(
-                self.authToken,
-                algorithms=["HS256"],
-                options={"verify_signature": False}
-            )
+            decoded = jwt.decode(self.authToken, algorithms=["HS256"], options={"verify_signature": False})
```
[the remark this answers is not in the document]
Comment (HiDiHo01) [reply python_frank_energie/authentication.py:74]: We don't have the key to verify signature

**Comment by HiDiHo01** [comment]
Looking good..this integration starts te become professional. Thanks
Let me know when it is ready to merge

**Comment by Veldkornet** [comment]
> Looking good..this integration starts te become professional. Thanks Let me know when it is ready to merge

Yep I think it's good. It works fine for me locally anyway with the HomeAssistant integartion updates as well.
